nservicebus/sagas/index.md

```diff
@@ -72,12 +72,17 @@ For more information about setting (requesting) the timeouts and handling them,
 When a message is received that could possibly be handled by a saga, and no existing saga can be found then that is handed by the [Saga Not Found](saga-not-found.md) feature.
 
 
-### Completing a Saga
+### Transactional behavior
 
-When completing a saga, it is important to take into considearation transport and persistence types. In case selected transport and persistence cannot participate in a shared transaction, sending messages along with marking saga as completed can result in saga completions and saga data cleanup despite outgoing messages failure. To avoid saga premature completion, the following options can be applied:
+Completing a saga is a destructive operation so transaction support of the selected transport and persitence must be considered to ensure correctness. If the persistence is able to participate in the same transaction as the incoming receive operation, either using DTC or by sharing the transports storage transaction (SQLServer transport), no futher action is needed.
 
-1. Enabling the [Outbox feature](/nservicebus/outbox) if supported by the chosen persistence.
-1. Setting a timeout or sending a message to self to invoke saga completion as a single operation that will be safely retried in case of a failure.
+If the persistence can't participate in the same transaction as the incoming receive operation, then an additional action is needed to avoid saga completion causing incorrect behavior. The problematic scenario is when sagas are being completed together with sending/publishing outgoing messages. In case of failure after the saga is completed, the outgoing messages may not be dispatched. However, when the incoming message is retried the completed saga is not found, which results in outgoing messages being lost.
+
+This issue can be avoided by:
+
+1. Enabling the [Outbox feature](/nservicebus/outbox), if supported by the chosen persistence.
+1. Making sure that no outgoing messages will be dispatched by completing the saga from a timeout.
```

Comment:
It's either completing from a timeout _or_ sending an explicit command to self. I'd like to keep the option of a command to self. Even though, in the end, both are messages 🙂